.github/workflows/publish.yml

```diff
@@ -0,0 +1,35 @@
+name: publish
+
+on:
+  push:
+    paths:
+      - "src/**"
+      - ".github/workflows/publish.yml"
+    branches:
+      - main
+  pull_request:
+    paths:
+      - "src/**"
+      - ".github/workflows/publish.yml"
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        machine: [windows-latest, macos-latest, ubuntu-20.04]
+        architecture: [x64]
+        include:
+          - machine: windows-latest
+            platform: win
+          - machine: macos-latest
+            platform: osx
+          - machine: ubuntu-20.04
```

Comment:
why not latest for linux?

Reply:
This is the latest LTS seems like a good choice. Although considering most the other dependencies are experimental 😆

Reply:
The version is built on is the minimum required to run. So if someone tried to run on ubuntu 18 they wouldn't be able to. But 20 and ahead they could. That's why I didn't pick latest for linux

Reply:
I would have picked ubuntu 18 but when looking at the github workers they said to use either 20 or 22

Looks like it's actually gone now 
https://github.blog/changelog/2022-08-09-github-actions-the-ubuntu-18-04-actions-runner-image-is-being-deprecated-and-will-be-removed-by-12-1-22/

Reply:
aaarrrgghh...

We can worry about this later, but eventually we'll definitely need Docker builds. E.g. Alpine definitely needs its own special builds, and likely Amazon Linux too. With Docker builds we can support older Ubuntu versions too...